src/duplex.rs

```diff
@@ -35,8 +37,29 @@ pub struct DuplexStream<Callback, Error> {
 
 /// Input proxy for transferring an input signal to a separate output callback to be processed as a duplex stream.
 pub struct InputProxy {
-    buffer: rtrb::Producer<f32>,
-    output_sample_rate: Arc<AtomicU64>,
+    producer: Option<fixed_resample::ResamplingProd<f32, MAX_CHANNELS>>,
+    receive_output_samplerate: rtrb::Consumer<u32>,
```

Comment:
Is using a ring buffer just to send a sample rate better than using an atomic?

Reply:
I'd need to have a separate signaling atomic for new values, this is similar to that but already self-contained, and we already have the dependency on `rtrb`, so I think it's fine